marquez_client/client_wo.py

```diff
@@ -0,0 +1,302 @@
+
+        payload = {
+            'ownerName': owner_name
+        }
+
+        if description:
+            payload['description'] = description
+
+        return self._backend.put(
+            self._url('/namespaces/{0}', namespace_name),
+            headers=_HEADERS,
+            json=payload
+        )
+
+    # Source API
+    def create_source(self, source_name, source_type, connection_url,
+                      description=None):
+        log.debug("create_source()")
+
+        MarquezWriteOnlyClient._check_name_length(source_name, 'source_name')
+        MarquezWriteOnlyClient._is_instance_of(source_type, SourceType)
+
+        MarquezWriteOnlyClient._is_valid_connection_url(connection_url)
+
+        payload = {
+            'type': source_type.value,
+            'connectionUrl': connection_url
+        }
+
+        if description:
+            payload['description'] = description
+
+        return self._backend.put(
+            self._url('/sources/{0}', source_name),
+            headers=_HEADERS,
+            json=payload)
+
+    # Datasets API
+    def create_dataset(self, namespace_name, dataset_name, dataset_type,
+                       physical_name, source_name, run_id,
+                       description=None, schema_location=None,
+                       fields=None, tags=None):
+        log.debug("create_dataset()")
+
+        MarquezWriteOnlyClient.\
+            _check_name_length(namespace_name, 'namespace_name')
+        MarquezWriteOnlyClient.\
+            _check_name_length(dataset_name, 'dataset_name')
+        MarquezWriteOnlyClient._is_instance_of(dataset_type, DatasetType)
+
+        if dataset_type == DatasetType.STREAM:
+            MarquezWriteOnlyClient.\
+                _is_none(schema_location, 'schema_location')
+
+        MarquezWriteOnlyClient._is_none(run_id, 'run_id')
+        MarquezWriteOnlyClient.\
+            _check_name_length(physical_name, 'physical_name')
+        MarquezWriteOnlyClient._check_name_length(source_name, 'source_name')
+
+        payload = {
+            'type': dataset_type.value,
+            'physicalName': physical_name,
+            'sourceName': source_name,
+            'runId': run_id,
+        }
+
+        if description:
+            payload['description'] = description
+
+        if fields:
+            payload['fields'] = fields
+
+        if tags:
+            payload['tags'] = tags
+
+        if schema_location:
+            payload['schemaLocation'] = schema_location
+
+        return self._backend.put(
+            self._url('/namespaces/{0}/datasets/{1}', namespace_name,
+                      dataset_name),
+            headers=_HEADERS,
+            json=payload
+        )
+
+    # Job API
+    def create_job(self, namespace_name, job_name, job_type,
+                   location=None, input_dataset=None,
+                   output_dataset=None, description=None, context=None):
+        log.debug("create_job()")
```

Comment:
_Minor_: Might want to remove?

Reply:
it's useful when troubleshooting. but can also be enabled through `format` in log config.